--- ./Assets/Core/Game/Domains/GamePlay/Simulation/Scripts/Physics/PhysicsSimulator.cs
+++ ./Assets/Core/Game/Domains/GamePlay/Simulation/Scripts/Physics/PhysicsSimulator.cs
@@ -756,7 +756,7 @@
             PhysicsBodyData bestHitData = default;
             float closestFraction = 1f;

-            void RayCastCallback(Box2D.NetStandard.Dynamics.Fixtures.Fixture fixture, Vector2 point, Vector2 normal, float fraction)
+            float RayCastCallback(Box2D.NetStandard.Dynamics.Fixtures.Fixture fixture, Vector2 point, Vector2 normal, float fraction)
             {
                 var body = fixture.Body;
                 if (body != null && body.UserData is PhysicsBodyData data)
@@ -768,6 +768,7 @@
                         bestHitData = data;
                     }
                 }
+                return closestFraction;
             }

             _world.RayCast(RayCastCallback, point1, point2);
