--- Assets/Core/Game/Domains/GamePlay/Presentation/Match/Scripts/TickProcessor/ClientMatchPresentationTickProcessor.cs
+++ Assets/Core/Game/Domains/GamePlay/Presentation/Match/Scripts/TickProcessor/ClientMatchPresentationTickProcessor.cs
@@ -42,6 +42,9 @@
         private readonly HandleSwapFieldCreatedNetEventsCommand _handleSwapFieldCreatedNetEventsCommand;
         private readonly HandleDeactivateSwapTalentNetEventsCommand _handleDeactivateSwapTalentNetEventsCommand;
         private readonly UpdateSwapFieldsTransformCommand _updateSwapFieldsTransformCommand;
+        private readonly HandleKOProjectileCreatedNetEventsCommand _handleKOProjectileCreatedNetEventsCommand;
+        private readonly HandleDeactivateKOTalentNetEventsCommand _handleDeactivateKOTalentNetEventsCommand;
+        private readonly HandleKOProjectHitPlayerNetEventsCommand _handleKOProjectHitPlayerNetEventsCommand;

         private int _tick;

@@ -73,6 +76,9 @@
             _handleSwapFieldCreatedNetEventsCommand = commandFactory.CreateCommandVoid<HandleSwapFieldCreatedNetEventsCommand>();
             _handleDeactivateSwapTalentNetEventsCommand = commandFactory.CreateCommandVoid<HandleDeactivateSwapTalentNetEventsCommand>();
             _updateSwapFieldsTransformCommand = commandFactory.CreateCommandVoid<UpdateSwapFieldsTransformCommand>();
+            _handleKOProjectileCreatedNetEventsCommand = commandFactory.CreateCommandVoid<HandleKOProjectileCreatedNetEventsCommand>();
+            _handleDeactivateKOTalentNetEventsCommand = commandFactory.CreateCommandVoid<HandleDeactivateKOTalentNetEventsCommand>();
+            _handleKOProjectHitPlayerNetEventsCommand = commandFactory.CreateCommandVoid<HandleKOProjectHitPlayerNetEventsCommand>();
         }

         public void FixedUpdate(float dt)
@@ -107,6 +113,9 @@
             _handlePreparationPhaseEndedNetEventsCommand.Execute();
             _handleSwapFieldCreatedNetEventsCommand.SetTick(_tick).Execute();
             _handleDeactivateSwapTalentNetEventsCommand.Execute();
+            _handleKOProjectileCreatedNetEventsCommand.Execute();
+            _handleDeactivateKOTalentNetEventsCommand.Execute();
+            _handleKOProjectHitPlayerNetEventsCommand.Execute();

             _updateEnvironmentSpringsTransformCommand.Execute();
             _updateEnvironmentTeleportPairsTransformCommand.Execute();
