--- ./Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/PlayerLockOnHeartTargetsChangedNetEventS2C.cs
+++ ./Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/PlayerLockOnHeartTargetsChangedNetEventS2C.cs
@@ -6,11 +6,6 @@
     public struct PlayerLockOnHeartTargetsChangedNetEventS2C : INetSerializable
     {
         public ushort PlayerId;
-        public FixedUnorderedList<ushort> LockedOnHeartIds;
-
-        public PlayerLockOnHeartTargetsChangedNetEventS2C(int maxHeartsIdsOnTarget)
-        {
-            PlayerId = 0;
-            LockedOnHeartIds = new FixedUnorderedList<ushort>(maxHeartsIdsOnTarget);
-        }
+        public FixedUnorderedList<ushort> LockedOnHeartIds; // it's a struct, parameterless constructor exists
