--- Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/MatchFullTickPacketS2C.cs
+++ Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/MatchFullTickPacketS2C.cs
@@ -34,6 +34,9 @@
         public FixedUnorderedList<PreparationPhaseEndedNetEventS2C> PreparationPhaseEndedNetEvents;
         public FixedUnorderedList<CreateSwapFieldNetEventS2C> CreateSwapFieldNetEvents;
         public FixedUnorderedList<DeactivateSwapTalentNetEventS2C> DestroySwapFieldNetEvents;
+        public FixedUnorderedList<KOProjectHitPlayerNetEventS2C> KOProjectHitPlayerNetEvents;
+        public FixedUnorderedList<CreateKOProjectileNetEventS2C> CreateKOProjectileNetEvents;
+        public FixedUnorderedList<DeactivateKOTalentNetEventS2C> DeactivateKOTalentNetEvents;

         public MatchFullTickPacketS2C()
         {
@@ -62,6 +65,9 @@
             PreparationPhaseEndedNetEvents = new FixedUnorderedList<PreparationPhaseEndedNetEventS2C>(maxCap.PreparationPhaseEndedNetEvents);
             CreateSwapFieldNetEvents = new FixedUnorderedList<CreateSwapFieldNetEventS2C>(maxCap.CreateSwapFieldNetEvents);
             DestroySwapFieldNetEvents = new FixedUnorderedList<DeactivateSwapTalentNetEventS2C>(maxCap.DestroySwapFieldNetEvents);
+            KOProjectHitPlayerNetEvents = new FixedUnorderedList<KOProjectHitPlayerNetEventS2C>(maxCap.KOProjectHitPlayerNetEvents);
+            CreateKOProjectileNetEvents = new FixedUnorderedList<CreateKOProjectileNetEventS2C>(maxCap.CreateKOProjectileNetEvents);
+            DeactivateKOTalentNetEvents = new FixedUnorderedList<DeactivateKOTalentNetEventS2C>(maxCap.DeactivateKOTalentNetEvents);
         }


@@ -102,6 +108,36 @@
             SerializedPreparationPhaseEndedEvents(writer);
             SerializedCreateSwapFieldNetEvents(writer);
             SerializedDestroySwapFieldNetEvents(writer);
+            SerializedKOProjectHitPlayerNetEvents(writer);
+            SerializedCreateKOProjectileNetEvents(writer);
+            SerializedDeactivateKOTalentNetEvents(writer);
+        }
+
+        private void SerializedKOProjectHitPlayerNetEvents(NetDataWriter writer)
+        {
+            writer.Put((byte)KOProjectHitPlayerNetEvents.Count);
+            foreach (var evt in KOProjectHitPlayerNetEvents.AsSpan())
+            {
+                evt.Serialize(writer);
+            }
+        }
+
+        private void SerializedCreateKOProjectileNetEvents(NetDataWriter writer)
+        {
+            writer.Put((byte)CreateKOProjectileNetEvents.Count);
+            foreach (var evt in CreateKOProjectileNetEvents.AsSpan())
+            {
+                evt.Serialize(writer);
+            }
+        }
+
+        private void SerializedDeactivateKOTalentNetEvents(NetDataWriter writer)
+        {
+            writer.Put((byte)DeactivateKOTalentNetEvents.Count);
+            foreach (var evt in DeactivateKOTalentNetEvents.AsSpan())
+            {
+                evt.Serialize(writer);
+            }
         }

         private void SerializedCreateSwapFieldNetEvents(NetDataWriter writer)
@@ -253,6 +289,39 @@
             DeserializedPreparationPhaseEndedEvents(reader);
             DeserializedCreateSwapFieldNetEvents(reader);
             DeserializedDestroySwapFieldNetEvents(reader);
+            DeserializedKOProjectHitPlayerNetEvents(reader);
+            DeserializedCreateKOProjectileNetEvents(reader);
+            DeserializedDeactivateKOTalentNetEvents(reader);
+        }
+
+        private void DeserializedKOProjectHitPlayerNetEvents(NetDataReader reader)
+        {
+            KOProjectHitPlayerNetEvents.Clear();
+            var count = reader.GetByte();
+            for (var i = 0; i < count; i++)
+            {
+                ref var evt = ref KOProjectHitPlayerNetEvents.AddAndGet();
+                evt.Deserialize(reader);
+            }
+        }
+
+        private void DeserializedCreateKOProjectileNetEvents(NetDataReader reader)
+        {
+            CreateKOProjectileNetEvents.Clear();
+            var count = reader.GetByte();
+            for (var i = 0; i < count; i++)
+            {
+                ref var evt = ref CreateKOProjectileNetEvents.AddAndGet();
+                evt.Deserialize(reader);
+            }
+        }
+
+        private void DeserializedDeactivateKOTalentNetEvents(NetDataReader reader)
+        {
+            DeactivateKOTalentNetEvents.Clear();
+            var count = reader.GetByte();
+            for (var i = 0; i < count; i++)
+            {
+                ref var evt = ref DeactivateKOTalentNetEvents.AddAndGet();
+                evt.Deserialize(reader);
+            }
         }

         private void DeserializedCreateSwapFieldNetEvents(NetDataReader reader)
