--- Assets/Core/Game/Domains/GamePlay/Presentation/Match/Features/KOProjectiles/Scripts/Mvc/KOProjectileController.cs
+++ Assets/Core/Game/Domains/GamePlay/Presentation/Match/Features/KOProjectiles/Scripts/Mvc/KOProjectileController.cs
@@ -10,6 +10,8 @@
         private readonly KOProjectileView _view;
         private readonly MatchKOProjectileModel _model;
         private readonly MatchPlayerModel _casterModel;
+
+        public KOProjectileView View => _view;

         public KOProjectileController(KOProjectileView view, MatchKOProjectileModel model, MatchPlayerModel casterModel)
         {
