--- ./Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/MatchFullTickPacketS2C.cs
+++ ./Assets/Core/Game/Domains/GamePlay/Shared/Scripts/S2CModels/MatchFullTickPacketS2C.cs
@@ -969,5 +969,25 @@
                 netEvent.Deserialize(reader);
             }
         }
+
+        private void SerializedPlayerLockOnHeartTargetsChangedNetEvents(NetDataWriter writer)
+        {
+            writer.Put((byte)PlayerLockOnHeartTargetsChangedNetEvents.Count);
+            foreach (var netEvent in PlayerLockOnHeartTargetsChangedNetEvents.AsSpan())
+            {
+                netEvent.Serialize(writer);
+            }
+        }
+
+        private void DeserializedPlayerLockOnHeartTargetsChangedNetEvents(NetDataReader reader)
+        {
+            PlayerLockOnHeartTargetsChangedNetEvents.Clear();
+            var count = reader.GetByte();
+            for (var i = 0; i < count; i++)
+            {
+                ref var netEvent = ref PlayerLockOnHeartTargetsChangedNetEvents.AddAndGet();
+                netEvent.Deserialize(reader);
+            }
+        }
     }
 }
