--- ./Assets/Core/Game/Domains/GamePlay/Simulation/Match/Scripts/NetworkManager/TickHandlers/PacketsObservers/MatchPlayerInputsPacketsHandler.cs
+++ ./Assets/Core/Game/Domains/GamePlay/Simulation/Match/Scripts/NetworkManager/TickHandlers/PacketsObservers/MatchPlayerInputsPacketsHandler.cs
@@ -36,6 +36,7 @@
         private readonly IUpdateSubscriptionService _updateSubscriptionService;
         private readonly ICommandFactory _commandFactory;
         private readonly ISimulationInputService _simulationInputService;
+        private readonly Core.Game.Domains.GamePlay.Simulation.Match.Scripts.Services.LockOnHeartTargetService.ILockOnHeartTargetService _lockOnHeartTargetService;

         private readonly CapacityDict<ushort, FixedUnorderedList<MatchPlayerInputPacketC2S>> _inputsPerPlayer;
         private readonly CapacityDict<ushort, MatchPlayerInputPacketC2S> _lastProcessedInputPerPlayer;
@@ -54,7 +55,8 @@

         public MatchPlayerInputsPacketsHandler(IServerNetworkManager networkManager, IMatchDataService matchDataService,
             SimulationGamePlayConfig gamePlayConfig, NetworkConfig networkConfig, INetEventsDataService iNetEventsDataService, IPhysicsSimulator physicsSimulator, IUpdateSubscriptionService updateSubscriptionService, ICommandFactory commandFactory,
-            IPlayersTalentsManager playersTalentsManager, IPlaybackRecorderService playerbackRecorderService, ISimulationInputService simulationInputService)
+            IPlayersTalentsManager playersTalentsManager, IPlaybackRecorderService playerbackRecorderService, ISimulationInputService simulationInputService,
+            Core.Game.Domains.GamePlay.Simulation.Match.Scripts.Services.LockOnHeartTargetService.ILockOnHeartTargetService lockOnHeartTargetService)
         {
             _networkManager = networkManager;
             _matchDataService = matchDataService;
@@ -69,6 +71,7 @@
             _playersTalentsManager = playersTalentsManager;
             _playerbackRecorderService = playerbackRecorderService;
             _simulationInputService = simulationInputService;
+            _lockOnHeartTargetService = lockOnHeartTargetService;

             var concurrentPlayers = networkConfig.MaxCap.ConcurrentPlayers;
             _inputsPerPlayer = new CapacityDict<ushort, FixedUnorderedList<MatchPlayerInputPacketC2S>>(concurrentPlayers);
@@ -191,6 +194,8 @@
             var playerId = playerModel.Id;
             _simulationInputService.SetPlayerInput(playerId, PlayerInputType.Shoot, isShootInputPressed);

+            _lockOnHeartTargetService.Process(processedTick, playerModel);
+
             var shootState = playerModel.Spaceship.Shoot;
             var isReadyToShoot = shootState.CooldownSecondsLeft == shootState.MaxCooldown;
