--- ./Assets/Core/Game/Domains/GamePlay/Simulation/Scripts/NetworkManager/NetEventsDataService.cs
+++ ./Assets/Core/Game/Domains/GamePlay/Simulation/Scripts/NetworkManager/NetEventsDataService.cs
@@ -1551,6 +1551,19 @@
             LogService.LogTopic($"Created AddActivateYearsOfPainTalentNetEventS2C for caster {casterPlayerId} in tick {tick}. Cooldown: {cooldownEndTick}, didHitEnemy: {didHitEnemy}, hitEnemyId {hitEnemyId}", LogTopicType.ServerSimulationEvents);
         }

+        public void AddPlayerLockOnHeartTargetsChangedNetEvent(int onTick, ushort playerId, Core.Scripts.Utils.CustomCollections.FixedUnorderedList<ushort> targetHeartIds)
+        {
+            var pkt = GetNextFreeFullTickPacket(onTick);
+            ref var netEvent = ref pkt.PlayerLockOnHeartTargetsChangedNetEvents.AddAndGet();
+            netEvent.PlayerId = playerId;
+            netEvent.LockedOnHeartIds.Clear();
+            for (int i = 0; i < targetHeartIds.Count; i++)
+            {
+                netEvent.LockedOnHeartIds.Add(targetHeartIds[i]);
+            }
+            LogService.LogTopic($"Created AddPlayerLockOnHeartTargetsChangedNetEvent for player {playerId} in tick {onTick} with {targetHeartIds.Count} hearts.", LogTopicType.ServerSimulationEvents);
+        }
+
         public void StartSavingPlayerEvents(ushort playerId)
         {
             var startSavingTick = Mathf.Max(0, _tickService.CurrentTick - _maxSavedTicks + 1);
